package Msg.Protocol;

enum GameToDBMsgId
{
	eGameToDBMsgIdMin = 2000;
	ePostSql = 2001; // 投递sql语句到db服

	eGameToDBMsgIdMax = 3000;
}

// sql语句类别
enum SqlType {
	NormalSql = 0; // 正常sql
	PlayerSql 	= 1; // 玩家sql
	GuidSql 	= 2;   // 门派sql
}

// sql目的库
enum SqlDBType {
	GameDB 	= 0; // 游戏库
	LogDB 	= 1;  // 日志库
}

// 单次sql语句
message SqlMsg
{
	required SqlType sqltype 	= 1 [default = NormalSql]; // sql语句类别
	required fixed64 sqlid 		= 2; // sql语句对应的玩家id/门派id....等等
	required string sql 		= 3; // sql文本列表
}

// 游戏服发送给db服的sql语句
message PostSql
{
	required fixed32 time 		= 1; // 该sql语句创建的时间
	required SqlDBType dbtype 	= 2; // sql将在哪个数据库上执行
	repeated SqlMsg sqls 		= 3; // sql列表
}